File: meme_inator_front/lib/features/profiles/domain/usecases/sync_profile_media_usecase.dart
import 'package:meme_inator_front/core/iusecase.dart';
import 'package:meme_inator_front/core/results.dart';
import 'package:meme_inator_front/features/profiles/domain/entities/profile_entity.dart';
import 'package:meme_inator_front/features/profiles/domain/interfaces/iprofile_repository.dart';
import 'package:uuid/uuid.dart';

class SyncProfileMediaInput {
  final Uuid userId;
  final Map<String, dynamic> mediaPayload;

  const SyncProfileMediaInput({
    required this.userId,
    required this.mediaPayload,
  });
}

class SyncProfileMediaUsecase
    extends IUseCase<SyncProfileMediaInput, ProfileEntity> {
  final IProfileRepository profileRepo;

  SyncProfileMediaUsecase(this.profileRepo);

  @override
  Future<Result<ProfileEntity>> execute({
    required SyncProfileMediaInput valObj,
  }) async {
    throw UnimplementedError('SyncProfileMediaUsecase.execute not implemented');
  }
}

File: meme_inator_back/apps/profiles/application/usecases/sync_profile_media_usecase.py
from typing import Dict, Any
from uuid import UUID
from apps.profiles.domain.entities.profile_entity import ProfileEntity
from apps.profiles.domain.usecases.isync_profile_media_usecase import ISyncProfileMediaUsecase
from apps.profiles.domain.irepositories.i_profile_repository import IProfileRepository
from core.results import Result

class SyncProfileMediaUsecase(ISyncProfileMediaUsecase):
    def __init__(self, profile_repo: IProfileRepository):
        self.profile_repo = profile_repo

    def execute(self, user_id: UUID, media_payload: Dict[str, Any]) -> Result[ProfileEntity]:
        return self.profile_repo.sync_media(user_id, media_payload)